### PR TITLE
As a sign owner in signage.chapman.edu, I would like the "expired" slides to be removed from the sign's details page.

### PR DESCRIPTION
**Pull request by ziegl107**
https://trello.com/c/FDYxnvmW/280-as-a-sign-owner-in-signage-chapman-edu-i-would-like-the-expired-slides-to-be-removed-from-the-sign-s-details-page

### CODE REVIEW

File: app/controllers/signs_controller.rb [hunk 1]
```diff
@@ -27,6 +27,9 @@ def index
   # GET /signs/1
   # GET /signs/1.json
   def show
+    @unexpired_slides = @sign.sign_slides.order(:order).eager_load(:slide).select do |slide|
+      !Slide.find(slide.slide_id).expired?
+    end
```
Comment (tatwell) [inline app/controllers/signs_controller.rb]: You're still working on revising this, yes?

Comment (ziegl107) [inline app/controllers/signs_controller.rb]: Yes, this is just a different way I found to do it

File: app/models/sign.rb [hunk 2]
```diff
@@ -38,6 +38,10 @@ def playable_slides
     slides.shown.active.approved.ordered
   end
 
+  def unexpired_slides
+    sign_slides.unexpired
```
Comment (tatwell) [inline app/models/sign.rb]: Does anyone have an explanation for why this is a `has_many sign_slides` association *and* a `has_many slides` association? There is a destroy dependency with `slides`. But is there a reason for both cases? This seems like an anti-pattern to me. But I'm not sure.

File: test/integration/send_slide_to_sign_test.rb [hunk 3]
```diff
@@ -49,6 +49,16 @@ class SendSlideToSignTest < Capybara::Rails::TestCase
     assert slide_is_on_sign?('Ross Slide', signs(:default))
   end
 
+  test "expired slides not displayed" do
+    sign_in users(:ross)
+    sign = signs(:default)
+    sign.slides << slides(:expired)
```
Comment (tatwell) [inline test/integration/send_slide_to_sign_test.rb]: Was there already an `expired` slide fixture?

Comment (ziegl107) [inline test/integration/send_slide_to_sign_test.rb]: Yes there was already an `expired` slide fixture

**Pull request merged**